l10n_br_fiscal/tests/test_fiscal_document_generic.py

```diff
@@ -291,6 +297,98 @@ def test_nfe_other_state(self):
+            line.with_company(empresa_lucro_presumido.id)._onchange_product_id_fiscal()
+            line.with_company(
+                empresa_lucro_presumido.id
+            )._onchange_commercial_quantity()
+            line.with_company(empresa_lucro_presumido.id)._onchange_ncm_id()
+            line.with_company(
+                empresa_lucro_presumido.id
+            )._onchange_fiscal_operation_id()
+            line.with_company(
+                empresa_lucro_presumido.id
+            )._onchange_fiscal_operation_line_id()
+            line.with_company(empresa_lucro_presumido.id)._onchange_fiscal_taxes()
+
+            self.assertEqual(
+                line.cfop_id.code,
+                "6404",
+                "Error to mapping CFOP 6404"
+                " for Revenda de Contribuinte p/ Fora do Estado.",
+            )
+
+            # ICMS
+            if line.product_id.icms_origin not in ICMS_ORIGIN_TAX_IMPORTED:
+                line.with_company(empresa_lucro_presumido.id)._onchange_fiscal_taxes()
+                self.assertEqual(
+                    line.icms_tax_id.name,
+                    "ICMS Cobrado Ant. por ST",
+                    "Error to mapping ICMS Cobrado Ant. por ST"
+                    " for Venda de Contribuinte p/ Fora do Estado.",
+                )
+                self.assertEqual(
+                    line.icms_cst_id.code,
+                    "60",
+                    "Error to mapping CST 60 from ICMS Cobrado Ant. por ST"
+                    " for Venda de Contribuinte p/ Fora do Estado.",
+                )
+
+    def test_nfe_other_state_st_6401(self):
+        """Testing NFe in another state with tax substitution."""
+        empresa_lucro_presumido = self.env.ref("l10n_br_base.empresa_lucro_presumido")
+        self.nfe_other_state_st_6401._onchange_document_serie_id()
+        self.nfe_other_state_st_6401._onchange_fiscal_operation_id()
+
+        for line in self.nfe_other_state_st_6401.fiscal_line_ids:
+            line.with_company(empresa_lucro_presumido.id)._onchange_product_id_fiscal()
+            line.with_company(
+                empresa_lucro_presumido.id
+            )._onchange_commercial_quantity()
+            line.with_company(empresa_lucro_presumido.id)._onchange_ncm_id()
+            line.with_company(
+                empresa_lucro_presumido.id
+            )._onchange_fiscal_operation_id()
+            line.with_company(
+                empresa_lucro_presumido.id
+            )._onchange_fiscal_operation_line_id()
+            line.with_company(empresa_lucro_presumido.id)._onchange_fiscal_taxes()
+
+            self.assertEqual(
+                line.cfop_id.code,
+                "6401",
+                "Error to mapping CFOP 6401"
+                " for Venda de Contribuinte p/ Fora do Estado.",
+            )
+
+            # ICMS
+            if line.product_id.icms_origin not in ICMS_ORIGIN_TAX_IMPORTED:
```

Comment:
`if line.product_id.icms_origin not in ICMS_ORIGIN_TAX_IMPORTED:`

Esse if é desnecessário